backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java

```diff
@@ -383,23 +382,25 @@ public String uploadFile(
     }
   }
 
-  public Resource loadFileAsResource(String artifactPath) throws ModelDBException {
+  public S3Object loadFileAsResource(String artifactPath) throws ModelDBException {
     LOGGER.trace("S3Service - loadFileAsResource called");
     try {
       if (doesObjectExist(bucketName, artifactPath)) {
         LOGGER.trace("S3Service - loadFileAsResource - resource exists");
         LOGGER.trace("S3Service - loadFileAsResource returned");
-        return new InputStreamResource(
-            s3Client.getObject(bucketName, artifactPath).getObjectContent());
+        return s3Client.getObject(bucketName, artifactPath);
```

Comment:
this throws .SdkClientException as well , let us add a catch for it

Reply:
i see a warning in the javadoc of this method 

```
Be extremely careful when using this method; the returned Amazon S3 object contains a direct stream of data from the HTTP connection. The underlying HTTP connection cannot be reused until the user finishes reading the data and closes the stream. Also note that if not all data is read from the stream then the SDK will abort the underlying connection, this may have a negative impact on performance. Therefore:

Use the data from the input stream in Amazon S3 object as soon as possible
Read all data from the stream (use GetObjectRequest.setRange(long, long) to request only the bytes you need)
Close the input stream in Amazon S3 object as soon as possible
If these rules are not followed, the client can run out of resources by allocating too many open, but unused, HTTP connections
```

can you ensure we are following the correct pattern? I do not see any close() being called here